wrappers/python/simtk/openmm/app/topology.py

```diff
@@ -404,3 +436,32 @@ def __init__(self, name, element, index, residue, id):
 
     def __repr__(self):
         return "<Atom %d (%s) of chain %d residue %d (%s)>" % (self.index, self.name, self.residue.chain.index, self.residue.index, self.residue.name)
+
+class Bond(tuple):
```

Comment:
namedtuple requires that all members are actually part of the tuple.  For backward compatibility, this has to be a two element tuple containing only the two atoms.  And namedtuple doesn't have a per-instance dict, so you can't then add extra fields to it.